llvm/0102-Clang-Cygwin-Enable-few-conditions-that-are-shared-w.patch

```diff
@@ -0,0 +1,183 @@
+From c95c2724dc567f878b3e1a33322077d7a6f9cd13 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Mateusz=20Miku=C5=82a?= <oss@mateuszmikula.dev>
+Date: Sat, 8 Feb 2025 01:13:06 +0100
+Subject: [PATCH 2/5] [Clang][Cygwin] Enable few conditions that are shared
+ with MinGW
+
+---
+ clang/lib/AST/ItaniumCXXABI.cpp       | 2 +-
+ clang/lib/AST/RecordLayoutBuilder.cpp | 2 +-
+ clang/lib/CodeGen/CodeGenModule.cpp   | 2 +-
+ clang/lib/CodeGen/ItaniumCXXABI.cpp   | 4 ++--
+ clang/lib/CodeGen/Targets/X86.cpp     | 2 +-
+ clang/lib/Driver/ToolChain.cpp        | 2 +-
+ clang/lib/Sema/SemaDecl.cpp           | 2 +-
+ clang/lib/Sema/SemaDeclCXX.cpp        | 4 ++--
+ clang/lib/Sema/SemaTemplate.cpp       | 8 ++++----
+ 9 files changed, 14 insertions(+), 14 deletions(-)
+
+diff --git a/clang/lib/AST/ItaniumCXXABI.cpp b/clang/lib/AST/ItaniumCXXABI.cpp
+index a1b2551419f5..d866f6023e14 100644
+--- a/clang/lib/AST/ItaniumCXXABI.cpp
++++ b/clang/lib/AST/ItaniumCXXABI.cpp
+@@ -236,7 +236,7 @@ public:
+ 
+   CallingConv getDefaultMethodCallConv(bool isVariadic) const override {
+     const llvm::Triple &T = Context.getTargetInfo().getTriple();
+-    if (!isVariadic && T.isWindowsGNUEnvironment() &&
++    if (!isVariadic && T.isOSCygMing() &&
```

Comment:
Random thought: maybe this change was wrong?  I'll try reverting this

Reply:
Very likely, I didn't test 32-bit binaries at all. I just assumed Cygwin would behave just like MinGW in this case.

Reply:
With this change backed out, clang can build itself on i686